devito/arch/compiler.py

```diff
@@ -396,8 +396,10 @@ def __init__(self, *args, **kwargs):
             self.cflags.remove('-std=c99')
             # Add flags for OpenMP offloading
             if language in ['C', 'openmp']:
-                # TODO: Need a generic -march setup
-                # self.cflags += ['-Xopenmp-target', '-march=sm_37']
+                # Autodetect GPU compute capability
+                if detect_cc():
+                    self.cflags += ['-Xopenmp-target', ''.join(['-march=sm_',
```

Comment:
why do you need `join`. just `'-march=sm_%s' % detect_cc()` should work no

Reply:
indeed...

Reply:
also, it's quite ugly, as you're calling `detect_cc` twice, relying on the assumption that it gets memorized

And, finally, we want `'-Xopenmp-target'` to always be added at least, no? so that we'll try to compile, even if the CC couldn't be detected for whatever reason.

Reply:
>And, finally, we want '-Xopenmp-target' to always be added at least, no? so that we'll try to compile, even if the CC couldn't be detected for whatever reason.

We were not using it before, why should we do now?